tests/child_chain/test_child_chain.py

```diff
@@ -32,7 +33,9 @@ def child_chain():
     # Create a block with those transactions
     child_chain.blocks[1] = Block([tx1, tx2])
 
-    return child_chain
+    yield child_chain
+
+    child_chain.event_listener.stop_all()
```

Comment:
Shouldn't stopping the events that the child chain is instantiated with come before `yield`?

Reply:
`yield` returns the child chain immediately (for use in tests), which starts event listeners. The `stop_all()` statement is just teardown after the tests have completed.